### PR TITLE
Look for files in a category as generator

### PR DESCRIPTION
**Pull request by nemobis**

### CODE REVIEW

File: etc/mediacounts-stats.py [hunk 1]
```diff
@@ -50,39 +56,56 @@ def log(msg):
...
 
 def process():
-    tsvfile = open(args.input)
-    tsvfilesize = os.path.getsize(args.input)
     csvfile = open(args.output, "w")
     writer = csv.writer(csvfile)
     rowwritten = False
 
+    for path in query():
+        # FIXME: Should search for sorted titles and continue seeking where it left.
```
Comment (nemobis) [inline etc/mediacounts-stats.py]: Which is what master is doing. Ouch.

**Comment by nemobis** [comment]
Cleaned up the diff now. Faster than my previous version was, but would break with a huge list of files.

**Comment by nemobis** [comment]
https://github.com/nemobis/wiki-tools/commit/478018a1dae5aec7b2197898a0039c82e2285cc5 should be useful for you too. :)

**Comment by hay** [comment]
Awesome, thanks for the PR!

**Comment by hay** [comment]
I've updated your code a bit so that the error is only given when the `-cat` flag is actually used. Otherwise, the error might be confusing.

**Comment by nemobis** [comment]
Hay Kranen, 30/03/2015 10:27:
> I've updated your code a bit so that the error is only given when the
> |-cat| flag is actually used. Otherwise, the error might be confusing.

Ok, sorry for not thinking about it.

**Comment by hay** [comment]
Not a problem!

**Pull request merged**